python/lsst/drp/tasks/wcsfit.py

```diff
@@ -0,0 +1,1019 @@
+            WCS-fitting object
+        inputCatalogRefs : `list`
+            List of DeferredDatasetHandles pointing to visit-level source
+            tables
+        """
+        for inputCatalogRef in inputCatalogRefs:
+            visit = inputCatalogRef.dataId['visit']
+
+            inputCatalog = inputCatalogRef.get(parameters={'columns': self.columns})
+            detectors = set(inputCatalog['detector'])
+
+            for detector in detectors:
+                detectorSources = inputCatalog[inputCatalog['detector'] == detector]
+
+                extensionIndex = self.extensionInfo[(self.extensionInfo['visit'] == visit)
+                                                    & (self.extensionInfo['detector'] == detector)].index[0]
+                sourceCat = detectorSources[self.sourceIndices[extensionIndex]]
+
+                d = {"x": sourceCat['x'].to_numpy(),
+                     "y": sourceCat['y'].to_numpy(),
+                     "xCov": sourceCat['xErr'].to_numpy()**2,
+                     "yCov": sourceCat['yErr'].to_numpy()**2,
+                     "xyCov": np.zeros(len(sourceCat))}
+                # TODO: add correct xyErr if DM-7101 is ever done
+
+                wcsf.setObjects(extensionIndex, d, 'x', 'y', ['xCov', 'yCov', 'xyCov'])
+
+    def _addRefObjects(self, wcsf):
+        """Add reference sources to the wcsfit.WCSFit object.
+
+        Parameters
+        ----------
+        wcsf : `wcsfit.WCSFit`
+            WCS-fitting object
+        """
+        extensionIndex = self.extensionInfo[self.extensionInfo['extensionType'] == 'REFERENCE'].index[0]
+
+        if self.config.useProperMotion:
+            wcsf.setObjects(extensionIndex, self.refObjects, 'ra', 'dec', ['raCov', 'decCov', 'raDecCov'],
+                            pmDecKey='decPM', pmRaKey='raPM', parallaxKey='parallax', pmCovKey='fullCov',
+                            pmCov=self.refCovariance)
+        else:
+            wcsf.setObjects(extensionIndex, self.refObjects, 'ra', 'dec', ['raCov', 'decCov', 'raDecCov'])
+
+    def _makeAfwWcs(self, mapDict, centerRA, centerDec, doNormalizePixels=False, xScale=1, yScale=1):
+        """Make an `lsst.afw.geom.SkyWcs` from a dictionary of mappings
+
+        Parameters
+        ----------
+        mapDict : `dict`
+            Dictionary of mapping parameters
+        centerRA : `lsst.geom.Angle`
+            RA of the tangent point
+        centerDec : `lsst.geom.Angle`
+            Declination of the tangent point
+        doNormalizePixels : `bool`
+            Whether to normalize pixels so that range is [-1,1]
+        xScale : `float`
+            Factor by which to normalize x-dimension. Corresponds to width of
+            detector
+        yScale : `float`
+            Factor by which to normalize y-dimension. Corresponds to height of
+            detector
+
+        Returns
+        -------
+        outWCS : `lsst.afw.geom.SkyWcs`
+            WCS constructed from the input mappings
```

Comment:
Looking at the underlying code, the WCS this returns does not have a FOCAL frame, as our camera convention assumes. Is that what `INTERMEDIATE` is supposed to be? If there isn't a FOCAL frame, we may have difficulty integrating these output WCSs with the rest of our tooling (e.g. for computing radial distortions).

Reply:
I believe the expected frames are the ones documented here: https://pipelines.lsst.io/modules/lsst.afw.cameraGeom/cameraGeom.html

Reply:
The names here, if I am reading it correctly, don't match the frame names in `jointcal`, so I don't think this can be right. I've been testing `WCSFit` in the DRP-RC2 pipeline, and the downstream tasks seem to work with this configuration of the wcs.